frontend/src/components/vue_canvas/instances/BoxInstance.ts

```diff
@@ -0,0 +1,342 @@
+import {InstanceBehaviour2D, Instance} from './Instance'
+import {InstanceContext} from './InstanceContext';
+import {v4 as uuidv4} from 'uuid';
+import {getContrastColor} from '../../../utils/colorUtils'
+import {MousePosition, point_is_intersecting_circle} from "../../../types/mouse_position";
+import {get_sequence_color} from '../../regular/regular_annotation'
+import {InstanceColor} from "../../../types/instance_color";
+import {LabelColourMap} from "../../../types/label_colour_map";
+import {ImageCanvasTransform} from "../../../types/CanvasTransform";
+import {InstanceImage2D} from "./InstanceImage2D";
+import {ImageLabelSettings} from "../../../types/image_label_settings";
+
+type BoxHoverPoints =
+  'x_min_y_min'
+  | 'x_max_y_min'
+  | 'x_min_y_max'
+  | 'x_max_y_max'
+  | 'not_intersecting_special_points'
+  | 'blocked'
+
+export class BoxInstance extends InstanceImage2D implements InstanceBehaviour2D {
+  public mouse_position: MousePosition;
```

Comment:
I'm a bit confused with this class. That seems like it also takes care of drawing and some other logic that is related to the annotation interface, and not to the instance itself, am I right?